admin/installer/test/test_cloudformation.py

```diff
@@ -0,0 +1,98 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Tests for ``flocker.admin.installer.cloudformation``.
+"""
+
+from os import walk
+from subprocess import CalledProcessError
+
+from hypothesis import given
+from hypothesis.strategies import integers, one_of
+
+from twisted.python.filepath import FilePath
+
+from flocker.testtools import TestCase, run_process
+
+from .. import MIN_CLUSTER_SIZE, MAX_CLUSTER_SIZE
+
+# A Hypothesis strategy for generating supported cluster size.
+valid_cluster_size = integers(min_value=MIN_CLUSTER_SIZE,
+                              max_value=MAX_CLUSTER_SIZE)
+
+# A Hypothesis strategy for generating unsupported cluster size.
+too_small_cluster_size = integers(min_value=0,
+                                  max_value=MIN_CLUSTER_SIZE-1)
+too_big_cluster_size = integers(min_value=MAX_CLUSTER_SIZE+1,
+                                max_value=100)
+invalid_cluster_size = one_of(too_small_cluster_size, too_big_cluster_size)
+
+
+def _get_cloudformation_full_path():
+    """
+    Get fully qualified pathname of cloudformation.py script.
+
+    :returns: Fully qualified pathname of cloudformation.py
+    :rtype: twisted.python.filepath.FilePath
+    """
+    root_path = b'/'
+    cloudformation_path_suffix = b'flocker/admin/installer'
+    cloudformation_file = b'cloudformation.py'
+    for root, dirs, files in walk(root_path):
+        if root.endswith(cloudformation_path_suffix) \
+           and cloudformation_file in files:
+            return FilePath("/".join((root, cloudformation_file)))
```

Comment:
This walks from the root trying to find something that looks like the right path?

This seems likely to bite people later who have two trees around (Jenkins slaves
certainly do)

Can this not be written in terms of __file__?